### ISSUE TITLE
Animate to position with offset?

### ISSUE THREAD

**Issue by nazywamsiepawel** [react-community/react-native-maps#307]
Hey all,

I'm working on a feature in which maps animates to center custom markers. There is a list of places at the bottom from which users select which place they want to highlight. I do the animation like that :

```
let location = place.googlePlace.geometry.location;
region.timing({
    latitude: location.lat,
    longitude: location.lng,
}).start();
```

Everything animates great but there is one issue - the point animated ends up in the centre (as expected) but I need to push it a bit up with some offset. My first native solution was to add some small value to location.lat but it doesn't work very well with animation that has to zoom in/out.

tl;dr question is - how to animate to a region on the map with an offset of y pixels?

Thanks!

**Comment by jrichardlai** [comment]
I would think that adding a lat and lng offset would do it, what is the issue that is happening? can you make a gif?

**Comment by xvalentino** [comment]
I am using a Custom Marker, which doesn't get rendered all of the way (it was getting cut off at the bottom). What I did is put padding at the bottom of the marker. This way I could modify the placement of the marker, but not actual lat/long values themselves. So the styling for my Image or alternate component looks something like: 

```
const styles = StyleSheet.create({
  marker: {
    paddingBottom: 10,
  },
});
```

Is this what you are looking for? @nazywamsiepawel

**Comment by arnaudambro** [comment]
Hello,
I actually have the same target : I have a marker on the center, and I would like to animate the map to see the marker at 30 points from the top of the screen.
![from](https://user-images.githubusercontent.com/31724752/48621516-02760f00-e9a4-11e8-8a97-1dca66f628d8.png)
![to](https://user-images.githubusercontent.com/31724752/48621518-02760f00-e9a4-11e8-8dab-7e7d903a3e6c.png)

I can achieve this by giving an offset to the latitude (for example `latitude - 0.4 * latitudeDelta`), but it is not working cross-screens, how would you do ?

**Comment by LucienPerouze** [comment]
The official docs recomands to use **mapPadding** for that :

See : [\<MapView /\> Component API](https://github.com/react-native-community/react-native-maps/blob/master/docs/mapview.md#mapview--component-api)

| Prop | Type | Default | Note |
|---|---|---|---|
| `mapPadding` | `EdgePadding` |  | Adds custom padding to each side of the map. Useful when map elements/markers are obscured. **Note** Google Maps only.

For example :
``
mapPadding={{top: 100, left: 0, right: 0, bottom: 0}}
``

**Comment by amirmatin** [comment]
@LucienPerouze worked like a charm

**Comment by ThanhNghia944** [comment]
thank you @LucienPerouze

**Comment by VehpuS** [comment]
> The official docs recomands to use **mapPadding** for that :
> 
> See : [<MapView /> Component API](https://github.com/react-native-community/react-native-maps/blob/master/docs/mapview.md#mapview--component-api)
> 
> Prop	Type	Default	Note
> `mapPadding`	`EdgePadding`		Adds custom padding to each side of the map. Useful when map elements/markers are obscured. **Note** Google Maps only.
> For example :
> `mapPadding={{top: 100, left: 0, right: 0, bottom: 0}}`

The problem with this approach is that it affects google's map elements (i.e. compass, google text on the bottom). Is there a ways to provide an offset to the marker (without changing zoom levels, as in "fitToCoordinates" with an edge offset.

**Comment by Appkidd** [comment]
@VehpuS Just hit this problem, too. To work around this I decided to offset the latitude value to emulate the effect of `edgePadding`. Initial tests on an iPhone 8 and iPhone 11 show that this handles well on different screen sizes. The amount you offset by will obviously depend on your zoom/altitude, but with altitude set at 2000 I found that `longitude - 0.0015` worked nicely.

**Comment by mvanroon** [comment]
A new prop that overrides the offset of the map elements (compass, logo's, legal, etc.) would solve this problem. This new prop should default to the value of `mapPadding`. E.g. `interfaceInsets`, `userInterfaceInsets` or `interfacePadding`.

Or it could be the other way around: introduce a new prop for offsetting the map that defaults to mapPadding, e.g. 'mapOffset` or 'cameraOffset`.

Thoughts?

**Comment by sunny-mittal** [comment]
I just had this same issue and did something like this:
```
const { northEast, southWest } = await mapRef.current.getMapBoundaries()
const latOffset = (northEast.latitude - southWest.latitude) * 0.1
mapRef.current.animateCamera({
  center: {
    latitude: pool.latitude_round - latOffset,
    longitude: pool.longitude_round,
  },
})
```
In my particular instance, I wanted to center the map about 10% of the screen height below where it would have gone without the offset and this worked for me. Some minor tweaks could make it work for other use cases but an advantage of this approach (for me) is that using the NE and SW coordinates, I automatically account for zoom. Hope it helps!

**Comment by urkin-dev** [comment]
@sunny-mittal Thanks it works, but when you rotate the map lat and lon are also changing and the center goes in wrong direction
